src/main/java/graphql/execution/ResolveType.java

```diff
@@ -19,11 +20,20 @@
 @Internal
 public class ResolveType {
 
-
     public GraphQLObjectType resolveType(ExecutionContext executionContext, MergedField field, Object source, ExecutionStepInfo executionStepInfo, GraphQLType fieldType, Object localContext) {
-        GraphQLObjectType resolvedType;
+        Assert.assertTrue(fieldType instanceof GraphQLInterfaceType || fieldType instanceof GraphQLUnionType,
+                () -> "The passed in fieldType MUST be an interface or union type : " + fieldType.getClass().getName());
+        TypeResolutionEnvironment env = buildEnv(executionContext, field, source, executionStepInfo, fieldType, localContext);
```

Comment:
We now ONLY take unions and interfaces.  Since the object type check is outside now.  This class is NOT general purpose and ONLY used above